src/govuk/components/pagination/pagination.yaml

```diff
@@ -0,0 +1,243 @@
+params:
+- name: previous
+  type: object
+  required: false
+  description: A link to the previous page, if there is one.
+  params:
+  - name: text
+    type: string
+    required: false
+    description: The link text to the previous page. Defaults to 'Previous page', where 'page' is visually hidden.
+  - name: visuallyHiddenText
+    type: string
+    required: false
+    description: The visually hidden label for the pagination item which will be applied to an `aria-label` and announced by screen readers on the previous page link.
+  - name: labelText
+    type: string
+    required: false
+    description: Optional label that goes underneath the previous page's link, providing further context for the user about where it goes.
+  - name: href
+    type: string
+    required: true
+    description: The previous page's URL.
+  - name: attributes
+    type: object
+    required: false
+    description: HTML attributes (for example data attributes) to add to the anchor.
+- name: next
+  type: object
+  required: false
+  description: A link to the next page, if there is one.
+  params:
+  - name: text
+    type: string
+    required: false
+    description: The link text to the next page. Defaults to 'Next page', where 'page' is visually hidden.
+  - name: visuallyHiddenText
+    type: string
+    required: false
+    description: The visually hidden label for the pagination item which will be applied to an `aria-label` and announced by screen readers on the next page link.
+  - name: labelText
+    type: string
+    required: false
+    description: Optional label that goes underneath the next page link providing further context for where it goes.
+  - name: href
+    type: string
+    required: true
+    description: The next page URL.
+  - name: attributes
+    type: object
+    required: false
+    description: HTML attributes (for example data attributes) to add to the anchor.
+- name: landmarkLabel
+  type: string
+  required: false
+  description: The label for the navigation landmark that wraps the pagination. Defaults to "results".
+- name: items
+  type: array
+  required: false
+  description: Array of link objects.
+  params:
+  - name: number
+    type: string
+    required: false
+    description: The pagination item text - usually a page number.
+  - name: visuallyHiddenText
+    type: string
+    required: false
+    description: The visually hidden label for the pagination item which will be applied to an `aria-label` and announced by screen readers on the pagination item link.
+  - name: href
+    type: string
+    required: false
+    description: The link URL.
+  - name: current
+    type: boolean
+    required: false
+    description: Set to 'true' to indicate the current page.
+  - name: ellipsis
+    type: boolean
+    required: false
+    description: For specifying if you want an ellipsis at a given point between numbers. If you set this option as `true`, any other options for the item are ignored.
```

Comment:
In this context I used it to mean "tell the macro that we want this item to just be an ellipsis instead of a link/number like most of the other pagination items".

The use case for ellipses is when you want to portray a large number of pages in a pagination without printing every single page in this element. There are details on the [website guidance preview](https://add-pagination--govuk-design-system-preview.netlify.app/components/pagination/#for-larger-numbers-of-pages) (although awkwardly at present it's using the wrong code example).